### PR TITLE
Fix undefined bug for template strings.

### LINKED ISSUE #40: The template literal for undefined not calculated correctly in props.
Given the following:

```
<a href={`${undefined}`} />
```

In an actual component, this is rendered as:

```
<a href="undefined" />
```

This is the default behaviour for such a template literal in JavaScript.

However in `getPropValue` this is returned as an empty string. 

It seems to be intended behaviour as there is also a test for this:
```
    it('should drop variables in template literals that are literally undefined', () => {
      const prop = extractProp('<div foo={`bar ${undefined}`} />');

      const expected = 'bar ';
      const actual = getPropValue(prop);

      assert.equal(expected, actual);
    });
```

However, the difference between the actual JSX / JavaScript behaviour and the result calculated here can lead to confusion when used in linters and tests.

### PR DESCRIPTION
**Pull request by evcohen**
Fixes #40 

cc: @AlmeroSteyn is this the expectation?

### CODE REVIEW

**Comment by AlmeroSteyn** [comment]
@evcohen Thank you! 

> template literals always toString directly, and null and undefined toString to "null" and "undefined" respectively.

The issue only pertained to `undefined` which now works perfectly as expected.

But if I apply the fix in my fork and run with `null` it still returns an empty string instead of `"null"`. 

Separate issue?

File: src/values/expressions/TemplateLiteral.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default function extractValueFromTemplateLiteral(value) {
     if (type === 'TemplateElement') {
       return raw + part.value.raw;
     } else if (type === 'Identifier') {
-      return part.name === 'undefined' ? raw : `${raw}{${part.name}}`;
+      return part.name === 'undefined' ? `${raw}${part.name}` : `${raw}{${part.name}}`;
```
Comment (ljharb) [inline src/values/expressions/TemplateLiteral.js]: This should check for both "undefined" or "null", perhaps?

Comment (AlmeroSteyn) [inline src/values/expressions/TemplateLiteral.js]: I initially tried this myself but for `null` the `part.name` is actually `undefined` and the `part.value.raw` throws a TypeError so it seems that something is happening higher up already.

**Comment by evcohen** [comment]
uh oh it looks like this may have been a breaking change. eslint-plugin-jsx-a11y tests are failing on v2.1.0

**Comment by ljharb** [comment]
Which tests? Is it possible the tests were incorrect?

**Comment by evcohen** [comment]
Yes that looks to be the case. Fixing now

**Pull request merged**